webui/module/Restore/src/Restore/Controller/RestoreController.php

```diff
@@ -588,7 +588,7 @@ private function buildSubtree()
           --$dnum;
           $items .= '{';
           $items .= '"id":"-' . $dir['pathid'] . '"';
-          $items .= ',"text":"' . preg_replace('/[\x00-\x1F\x7F]/', '', str_replace('"', '\"', $dir["name"])) . '"';
+          $items .= ',"text":"' . preg_replace('/[\x00-\x1F\x7F]/', '', str_replace('"', '\"', str_replace('\\', '\\\\', $dir["name"]))) . '"';
```

Comment:
that's the same three substitutions here and on line 611.
Don't you think this deserves its own function?

I'd also merge the two calls to `str_replace()` with a single call to `strtr()` with the second signature (i.e. that one with the `array $replace_pairs`.

Reply:
Good suggestion.

Reply:
Adapted.